qiskit/aqua/algorithms/single_sample/amplitude_estimation/ae_base.py

```diff
@@ -51,14 +57,48 @@ def a_factory(self, a_factory):
 
     @property
     def q_factory(self):
-        """ returns q factory """
-        return self._q_factory
+        """
+        Get the q_factory. If the q_factory is not set, we try to build it from the a_factory.
+        If the a_factory is not set as well, None is returned.
+
+        Returns:
+            QFactory: returns the current Q factory of the algorithm
+        """
+        if self._q_factory is not None:
+            return self._q_factory
+
+        if self._a_factory is not None:
+            return QFactory(self._a_factory, self.i_objective)
+
+        return None
 
     @q_factory.setter
     def q_factory(self, q_factory):
         """ sets q factory """
         self._q_factory = q_factory
 
+    @property
+    def i_objective(self):
+        """
+        Get the index of the objective qubit. If i_objective is not set, we check if the
+        q_factory is set and return the index specified there. If the q_factory is not defined,
+        the index equals the number of qubits of the a_factory minus one. If also the
+        a_factory is not set, return None.
+
+        Returns:
+            int: the index of the objective qubit
+        """
+        if self._i_objective is not None:
+            return self._i_objective
+
+        if self._q_factory is not None:
+            return self._q_factory.i_objective
+
+        if self._a_factory is not None:
+            return self.a_factory.num_target_qubits - 1
+
+        return None
+
     def check_factories(self):
         """
         Check if a_factory has been set, and set q_factory if it hasn't been
```

Comment:
Is this still correct?

Reply:
The `check_factories` can be removed, the algorithms can do these two lines of codes each.

Reply:
I mean that the comment says " and set q_factory if it hasn't been" but it doesn't look like we do that here anymore?

Reply:
That's correct, but we also don't need the whole check_factories anymore. Before, it checked the A and Q operators and the objective qubit index, but now, if it only checks for the A factory we don't really need an extra function anymore